File: docs/src/test/java/org/springframework/security/docs/servlet/authentication/mfawhencustomconditions/CustomizerAuthorizationManagerFactoryConfiguration.java
package org.springframework.security.docs.servlet.authentication.mfawhencustomconditions;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.authorization.AuthorizationManagerFactories;
import org.springframework.security.config.Customizer;

@Configuration(proxyBeanMethods = false)
class CustomizerAuthorizationManagerFactoryConfiguration {

	// tag::customizer[]
	@Bean
	Customizer<AuthorizationManagerFactories.AdditionalRequiredFactorsBuilder<Object>> additionalRequiredFactorsCustomizer() {
		return (builder) -> builder.when((auth) -> "admin".equals(auth.getName()));
	}
	// end::customizer[]

}

File: docs/src/test/kotlin/org/springframework/security/kt/docs/servlet/authentication/mfawhencustomconditions/CustomizerAuthorizationManagerFactoryConfiguration.kt
package org.springframework.security.kt.docs.servlet.authentication.mfawhencustomconditions

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.authorization.AuthorizationManagerFactories
import org.springframework.security.config.Customizer

@Configuration(proxyBeanMethods = false)
internal class CustomizerAuthorizationManagerFactoryConfiguration {

    // tag::customizer[]
    @Bean
    fun additionalRequiredFactorsCustomizer(): Customizer<AuthorizationManagerFactories.AdditionalRequiredFactorsBuilder<Any>> {
        return Customizer { builder -> builder.`when` { auth -> "admin" == auth.name } }
    }
    // end::customizer[]

}
